src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/DependencyInjectionSpecificationTests.cs

```diff
@@ -761,6 +761,28 @@ public void ClosedServicesPreferredOverOpenGenericServices()
             Assert.IsType<FakeService>(service);
         }
 
+        // Reproduces https://github.com/dotnet/runtime/issues/79938
+        [Fact]
+        public void ResolvingEnumerableContainingOpenGenericServiceUsesCorrectSlot()
```

Comment:
Does this have the same coverage as the previous attempt's test? https://github.com/dotnet/runtime/pull/68053/files#diff-e21ef8173802b1f4eb5f44f0d27bc3a72e8662924e63f14d4dea503814f3f988

Reply:
In some ways they are testing the same scenario, but this one doesn't rely on container validation to trigger it. I think it would be good to retain both cases. Thoughts?

Reply:
The other test (which never made it in since the PR was closed without commit) and the test below both call `BuildServiceProvider()` with open and closed generics but the other test also specified ServiceProviderOptions.ValidateOnBuild with `true` and `false`. Should that test also be added here?

Reply:
@steveharter my bad I didn't realize that the other test was not merged. I think it makes sense to port it over.

Reply:
Done!